src/daemon/netlink.c

```diff
@@ -40,6 +51,54 @@ struct lldpd_netlink {
 	struct interfaces_address_list *addresses;
 };
 
+
+static int
+netlink_socket_set_buffer_sizes(int s)
+{
+	int sndbuf = RTNL_SND_BUFSIZE;
+	int rcvbuf = RTNL_RCV_BUFSIZE;
+	int read = 0, size = 0;
+
+	if (setsockopt(s, SOL_SOCKET, SO_SNDBUF, &sndbuf, sizeof(sndbuf)) < 0) {
+		log_warn("netlink", "unable to set SO_SNDBUF to '%d'");
+		return -1;
+	}
+
+	if (setsockopt(s, SOL_SOCKET, SO_RCVBUF, &rcvbuf, sizeof(rcvbuf)) < 0) {
+		log_warn("netlink", "unable to set SO_RCVBUF to '%d'");
+		return -1;
+	}
```

Comment:
What's important is the receive buffer, right? We never send anything I think. This would half the memory usage needed. See my general comment about this.